### PR TITLE
docs: update CODEOWNERS review process

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

**Review by tgraf — APPROVED** [review]

File: Documentation/contributing/development/contributing_guide.rst
Comment (pchaigno) [inline Documentation/contributing/development/contributing_guide.rst]:
```suggestion
   so make sure to bookmark it. If a PR is assigned to one or more
   committers, they are expected to review that PR. Once they have
```

File: Documentation/contributing/development/contributing_guide.rst
Comment (pchaigno) [inline Documentation/contributing/development/contributing_guide.rst]: The documentation test is failing because we don't have that word in the dictionary yet.

**Review by pchaigno — COMMENTED** [review]
:tada:

**Review by joestringer — APPROVED** [review]
Modulo Paul's comments, LGTM.

**Review by qmonnet — APPROVED** [review]
Looks good!

**Review by pchaigno — APPROVED** [review]

**Review by sayboras — APPROVED** [review]
LGTM :+1: 

Keen to hear the rationale behind, as well as how can committer (like myself) know whom I should assign to :).

**Comment by pchaigno** [comment]
> Keen to hear the rationale behind, as well as how can committer (like myself) know whom I should assign to :).

@sayboras The assignments are automatically done by a bot from the CODEOWNERS file, as follows:
1. A team is assigned based on the CODEOWNERS (same as before).
2. The GitHub load balance feature is configured to pick one person from that team and require a review from that person.
3. Maintainer's little helper, a Cilium bot, goes through the list of reviewers and assign them to the pull request. It will only assign specific persons from the teams, not the entire teams.
4. The assignees review (that :heavy_check_mark: the review from them and from their teams) and remove their assignment from the pull request.
5. If an assignee doesn't feel comfortable reviewing the pull request, or they did review but want another opinion from the team, they can request a new review from someone else in the team.

For each team, we can exclude specific people from the review assignments, for example if they only want to receive notifications and/or don't feel ready to provide reviews yet.

The rationale for this change is to allow for better load balancing of review workloads across the team. It also gives you a way to list separately pull requests for which you are expected to provide a review from pull requests for which you are just receiving a notification (because your team needs to provide a review but you haven't been assigned specifically).

**Comment by sayboras** [comment]
> The rationale for this change is to allow for better load balancing of review workloads across the team. It also gives you a way to list separately pull requests for which you are expected to provide a review from pull requests for which you are just receiving a notification (because your team needs to provide a review but you haven't been assigned specifically).

It's awesome and thanks for your detailed explaination. Seems like I can _borrow_ this idea at work as well, just found out about this [page](https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/managing-code-review-assignment-for-your-team)

**Pull request merged**
